### PR TITLE
add SHACL generation

### PR DESCRIPTION
**Pull request by armintaenzertng**
generate a SHACL file `model_shacl.ttl` from the properties of classes.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -24,6 +24,12 @@ def get_args():
         help="Experimental! Generate RDF in turtle format",
     )
 
+    argparser.add_argument(
+        "--gen-shacl",
```
Comment (zvr) [inline main.py:28]: I don't think a new option is needed; we still generate RDF turtle -- SHACL is the way to express the restrictions in this format.

File: spec_parser/utils.py [hunk 2]
```diff
@@ -405,6 +438,31 @@ def _gen_rdf(self, g: rdflib.Graph) -> None:
         g.add((cur, RDFS.comment, Literal(self.description)))
         g.add((cur, NS0.term_status, Literal(self.metadata.get("Status")[0])))
 
+    def _gen_shacl(self, g: rdflib.Graph) -> None:
+
+        class_uri = URIRef(self.metadata["id"][0])
+        shape_uri = URIRef(self.metadata["id"][0] + "Shape")
```
Comment (zvr) [inline spec_parser/utils.py:444]: There is no need to generate separate "shape" nodes. By using the same node (as the one that is a `class`), you can skip the `targetClass` property.

**Review by zvr — CHANGES_REQUESTED** [review]

**Comment by zvr** [comment]
Copy-pasting an example to illustrate my comment about not having extra nodes.

Instead of
```
ex:Person
	a owl:Class .

ex:PersonShape
	a sh:NodeShape ;
	sh:targetClass ex:Person ;
	sh:property [
		sh:path ex:hasFather ;
		sh:maxCount 1 ;
		sh:class ex:Person ;
	] .
```

do it all in one like:
```
ex:Person
	a owl:Class, sh:NodeShape ;
	sh:property [
		sh:path ex:hasFather ;
		sh:maxCount 1 ;
		sh:class ex:Person ;
	] .
```

**Comment by armintaenzertng** [comment]
Thanks for the helpful input! I believe that it's best to merge this PR with the other one, then! :)

**Comment by armintaenzertng** [comment]
closing this as it is now contained in #36

**Pull request closed without merging**